peer/peer.go

```diff
@@ -0,0 +1,1942 @@
+}
+
+// Id returns the peer id.
+func (p *Peer) Id() int32 {
+	return p.id
+}
+
+// NA returns the peer network address.
+func (p *Peer) NA() *wire.NetAddress {
+	return p.na
+}
+
+// Addr returns the peer address.
+func (p *Peer) Addr() string {
+	return p.addr
+}
+
+// Filter returns the peer bloom filter.
+func (p *Peer) Filter() *bloom.Filter {
+	return p.filter
+}
+
+// Inbound returns whether the peer is inbound.
+func (p *Peer) Inbound() bool {
+	return p.inbound
+}
+
+// Persistent returns whether the peer is persistent.
+func (p *Peer) Persistent() bool {
+	return p.persistent
+}
+
+// RetryCount returns the number of times to retry peer connection.
+func (p *Peer) RetryCount() int64 {
+	return p.retryCount
+}
+
+// RequestQueue returns the peer request queue.
+func (p *Peer) RequestQueue() []*wire.InvVect {
+	return p.requestQueue
+}
+
+// SetRequestQueue sets the peer request queue to the given queue.
+func (p *Peer) SetRequestQueue(requestQueue []*wire.InvVect) {
+	p.requestQueue = requestQueue
+}
+
+// QueueRequest queues a peer request.
+func (p *Peer) QueueRequest(iv *wire.InvVect) {
+	p.requestQueue = append(p.requestQueue, iv)
+}
+
+// LastAnnouncedBlock returns the last announced block of the peer.
+func (p *Peer) LastAnnouncedBlock() *wire.ShaHash {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastAnnouncedBlock
+}
+
+// LastPingNonce returns the last ping nonce of the peer.
+func (p *Peer) LastPingNonce() uint64 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastPingNonce
+}
+
+// LastPingTime returns the last ping time of the peer.
+func (p *Peer) LastPingTime() time.Time {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastPingTime
+}
+
+// LastPingMicros returns the last ping micros of the peer.
+func (p *Peer) LastPingMicros() int64 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastPingMicros
+}
+
+// VersionKnown returns the whether or not the version of a peer is known
+// locally.  It is safe for concurrent access.
+func (p *Peer) VersionKnown() bool {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.versionKnown
+}
+
+// ProtocolVersion returns the peer protocol version in a manner that is safe
+// for concurrent access.
+func (p *Peer) ProtocolVersion() uint32 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.protocolVersion
+}
+
+// ContinueHash returns the continue hash.
+// It is safe for concurrent access.
+func (p *Peer) ContinueHash() *wire.ShaHash {
+	p.relayMtx.Lock()
+	defer p.relayMtx.Unlock()
+
+	return p.continueHash
+}
+
+// SetContinueHash sets the continue hash.
+// It is safe for concurrent access.
+func (p *Peer) SetContinueHash(hash *wire.ShaHash) {
+	p.relayMtx.Lock()
+	defer p.relayMtx.Unlock()
+
+	p.continueHash = hash
+}
+
+// RelayTxDisabled returns whether or not relaying of transactions is disabled.
+// It is safe for concurrent access.
+func (p *Peer) RelayTxDisabled() bool {
```

Comment:
This is another thing that is not unique to the peer and should be in the aforementioned data struct in the block manager.